internal/notifier/util.go

```diff
@@ -59,10 +59,29 @@ func formatNameAndDescription(event eventv1.Event) (string, string) {
 // involved object kind and name.
 func generateCommitStatusID(providerUID string, event eventv1.Event) string {
 	uidParts := strings.Split(providerUID, "-")
-	id := fmt.Sprintf("%s/%s/%s", event.InvolvedObject.Kind, event.InvolvedObject.Name, uidParts[0])
+	metadataLabelsSuffix := createMetadataLabelsSuffix(event)
+	id := fmt.Sprintf("%s/%s/%s%s", event.InvolvedObject.Kind, event.InvolvedObject.Name, uidParts[0], metadataLabelsSuffix)
 	return strings.ToLower(id)
 }
 
+// createMetadataLabelsSuffix returns a concated string depending on app.kubernetes.io/,
+// prefixed common labels https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/
+func createMetadataLabelsSuffix(event eventv1.Event) string {
+	metadataLabels := make([]string, 0)
+	prefixedMetadataLabelsSuffix := ""
+	for labelKey, labelValue := range event.Metadata {
+		if strings.Contains(labelKey, "app.kubernetes.io/") {
+			metadataLabels = append(metadataLabels, labelValue)
+		}
+	}
```

Comment:
This code doesn't assure a certain ordering in how the final string is constructed. You will also notice this when running the tests multiple times as they will fail occasionally.

I don't think we should implement such a non-deterministic behaviour as people may want to parse the string.

Reply:
sure, I am going to implement an string sorted keys construction.